### PR TITLE
CNF-14962: Initial doc for cluster provisioning

### PR DESCRIPTION
**Pull request by Missxiaoguo**

### CODE REVIEW

File: docs/cluster-provisioning.md [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+      provisioningStatus:
+        provisioningDetails: Cluster has installed and configured successfully
+        provisioningState: fulfilled
+    ```
+
+## Immutable ClusterInstance ##
+Once cluster installation has started (indicated by the `ClusterProvisioned` condition being InProgress), only the `extraLabels` and `extraAnnotations` fields can be modified in the ProvisioningRequest. Any changes to other immutable fields will cause the `ClusterInstanceRendered` condition to fail.
+
+## Monitoring Process
+To watch the IMS controller logs,
```
Comment (irinamihai) [inline docs/cluster-provisioning.md:265]: Maybe `:` instead of `,` at the end?

**Review by irinamihai — COMMENTED** [review]
This looks great. Thank you! I have just a couple of suggestions.

File: docs/cluster-provisioning.md [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+  - lastTransitionTime: "2024-10-19T00:00:06Z"
+    message: The cluster template validation succeeded
+    reason: Completed
+    status: "True"
+    type: ClusterTemplateValidated
+```
+
+## ProvisioningRequest CR
+
+A cluster-scoped CR managed by the IMS operator, providing all neccessary parameters for provisioning a cluster. An example of ProvisioningRequest can be found [here](../config/samples/v1alpha1_provisioningrequest.yaml).
```
Comment (browsell) [inline docs/cluster-provisioning.md:63]: nit, replace IMS with O-Cloud Manager throughout

File: docs/cluster-provisioning.md [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+6. Wait for hardware provisioning to complete. Once it completes, the plugin sends the allocated node infomation (BMC secret, BMC url, interface Mac addressess, etc.) in the NodePool status. IMS retrieves this data and update the rendered ClusterInstance CR with the retrieved data.
+7. Create the rendered ClusterInstance to kick off cluster installation. The SiteConfig operator consumes the CR and starts the installation.
+
+    Example status:
+
+    ```console
+    status:
+      clusterDetails:
+        clusterProvisionStartedAt: "2024-10-19T00:27:31Z"
+        name: cnfdf20
```
Comment (browsell) [inline docs/cluster-provisioning.md:160]: nit, use generic name

File: docs/cluster-provisioning.md [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+    reason: Completed
+    status: "True"
+    type: ClusterTemplateValidated
+```
+
+## ProvisioningRequest CR
+
+A cluster-scoped CR managed by the O-Cloud Manager, providing all neccessary parameters for provisioning a cluster. An example of ProvisioningRequest can be found [here](../config/samples/v1alpha1_provisioningrequest.yaml).
+
+The `spec` fields include:
```
Comment (bartwensley) [inline docs/cluster-provisioning.md:70]: Add a link to the CRD?

File: docs/cluster-provisioning.md [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+## Provisioning process walkthrough
+
+The O-Cloud Manager orchestrates the cluster provisioning process, which is initiated by creating a `ProvisioningRequest` CR. Below is the general flow of the provisioning process:
+
+**The general success workflow**:
+ProvisioningRequestValidated -> ClusterInstanceRendered -> ClusterResourcesCreated -> HardwareTemplateRendered -> HardwareProvisioned -> ClusterProvisioned -> ConfigurationApplied
+
+**Steps are**:
+
+1. O-Cloud Manager first validates the ProvisioningReqeust CR, including but not limited to:
```
Comment (bartwensley) [inline docs/cluster-provisioning.md:101]: spelling: ProvisioningReqeust

**Review by bartwensley — COMMENTED** [review]
Looks great Angie - just a couple minor comments.

File: docs/cluster-provisioning.md (hunk 4 above)
Comment (Missxiaoguo) [reply docs/cluster-provisioning.md:70]: Sure

**Pull request merged**